### ISSUE TITLE
Enable Pretty Printing in Logging

### ISSUE THREAD

**Issue by alzimmermsft** [Azure/azure-sdk-for-java#6439]
Currently, `HttpLoggingPolicy` has a constructor which takes in a flag to allow pretty printing of JSON bodies. This value is always `false` as it is an internal constructor which always has `false` passed. `HttpLogOptions` should have a flag property added which can turn this functionality on and off, to maintain functionality with previously released versions of Azure Core it should default to `false` unless overwhelmingly decided for it to be on by default. In addition to this flag investigation should be done if the same type of functionality should be added for XML bodies.

**Comment by sima-zhu** [comment]
Here is how it looks like when enable the prettyJSON:
```xml
[main] INFO com.azure.storage.queue.implementation.QueuesImpl$QueuesService.create - --> PUT https://teststorage.queue.core.windows.net/queueapitestsenqueuemessage6443101ed4f7b0ee840
[main] INFO com.azure.storage.queue.implementation.QueuesImpl$QueuesService.create - Date:Thu, 21 Nov 2019 00:29:50 GMT
Authorization:REDACTED
Content-Length:0
x-ms-version:2019-02-02
x-ms-meta-:REDACTED
x-ms-client-request-id:fafcc5c9-c957-4286-a384-18f3d6b3c375
User-Agent:azsdk-java-azure-storage-queue/12.1.0-beta.1 (1.8.0_221; Windows 10 10.0)

[parallel-3] INFO com.azure.storage.queue.implementation.MessagesImpl$MessagesService.enqueue - 106-byte body:%n<?xml version='1.0' encoding='UTF-8'?><QueueMessage><MessageText>test message</MessageText></QueueMessage>
[parallel-3] INFO com.azure.storage.queue.implementation.MessagesImpl$MessagesService.enqueue - --> END POST
[parallel-3] INFO com.azure.storage.queue.implementation.MessagesImpl$MessagesService.enqueue - <-- 201 https://teststorage.queue.core.windows.net/queueapitestsenqueuemessage6443101ed4f7b0ee840/messages (7 ms, 407-byte body)
[parallel-3] INFO com.azure.storage.queue.implementation.MessagesImpl$MessagesService.enqueue - Date:Tue, 15 Oct 2019 19:19:38 GMT
Server:Windows-Azure-Queue/1.0 Microsoft-HTTPAPI/2.0
Content-Length:407
Transfer-Encoding:chunked
x-ms-version:2018-03-28
Content-Type:application/xml
retry-after:0
x-ms-request-id:2f7e81db-f003-0082-1d8d-839126000000
```

**Comment by sima-zhu** [comment]
Here is the appConfig log when enable the prettyJson:
```
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - --> PUT http://localhost:8080/kv/key26354480067
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - Date:Thu, 21 Nov 2019 00:38:30 GMT
Content-Length:77
x-ms-return-client-request-id:true
If-None-Match:null
Accept:application/vnd.microsoft.azconfig.kv+json
Authorization:REDACTED
x-ms-content-sha256:REDACTED
Host:REDACTED
Content-Type:application/json
If-Match:"badETag"
x-ms-client-request-id:c7562b13-fc49-451c-9b47-8eaa49615157
User-Agent:azsdk-java-UnknownName/UnknownVersion (1.8.0_221; Windows 10 10.0)

[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - 77-byte body:%n{
  "key" : "key26354480067",
  "value" : "myNewValue",
  "etag" : "badETag",
  "locked" : false
}
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - --> END PUT
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - <-- 412 http://localhost:8080/kv/key26354480067 (26 ms, 0-byte body)
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - Date:Thu, 21 Nov 2019 00:37:30 GMT
```

**Comment by sima-zhu** [comment]
appConfig when disable the pretty JSON:
```
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - --> PUT http://localhost:8080/kv/key26354480067
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - Date:Thu, 21 Nov 2019 00:40:36 GMT
Content-Length:77
x-ms-return-client-request-id:true
If-None-Match:null
Accept:application/vnd.microsoft.azconfig.kv+json
Authorization:REDACTED
x-ms-content-sha256:REDACTED
Host:REDACTED
Content-Type:application/json
If-Match:"badETag"
x-ms-client-request-id:7adfb3f9-ee82-4ee0-be13-5913936c39fe
User-Agent:azsdk-java-UnknownName/UnknownVersion (1.8.0_221; Windows 10 10.0)

[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - 77-byte body:%n{"key":"key26354480067","value":"myNewValue","etag":"badETag","locked":false}
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - --> END PUT
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - <-- 412 http://localhost:8080/kv/key26354480067 (17 ms, 0-byte body)
[main] INFO com.azure.data.appconfiguration.ConfigurationService.setKey - Date:Thu, 21 Nov 2019 00:37:30 GMT

```

**Comment by alzimmermsft** [comment]
Fixed in PR #8773
